### ISSUE TITLE
DEBUG NSP session stay on pause(F6) and does not switch to Continue(F5)

### ISSUE THREAD

**Issue by R-github-sup** [IBM/db2forzosdeveloperextension-about#89]
## Development environment where the bug occurred

- Db2 Developer Extension version:
- Editor platform
  - [X] Visual Studio Code
  - [ ] Eclipse Theia
- Editor platform version: 1.7.2
- Operating system on which your editor runs (for example, Windows 10 2004 or MacOS Catalina 10.15.7): Windows 10
- Java Version (Run `java -version` and paste the details here): 
- Db2 for z/OS version (including function level for Db2 12): Db2 FL9 
- [Log files](https://ibm.github.io/db2forzosdeveloperextension-about/docs/troubleshooting/opening-an-issue-and-providing-logs) attached?: Yes/No No

## Problem description

Detailed steps for reproducing the problem:

1. First step

### Observed behavior

-

### Expected behavior

-

**Comment by katsoohoo** [comment]
Hi @R-github-sup,

Thanks for reporting an issue. As I cannot recreate the issue, can you provide [extension logs](https://ibm.github.io/db2forzosdeveloperextension-about/docs/troubleshooting/opening-an-issue-and-providing-logs) for when you hit this issue?

**Comment by R-github-sup** [comment]
Hi Katherine,
Attached the extension log.
Thanks for the support.


Rifat Hakim
DB2 for z/OS systems programmer
New York City Office of Technology & Innovation
718-804-6912
***@***.******@***.***>

[Graphical user interface, application  Description automatically generated]

From: Katherine Soohoo ***@***.***>
Sent: Monday, August 29, 2022 1:43 PM
To: IBM/db2forzosdeveloperextension-about ***@***.***>
Cc: Hakim, Rifat ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [IBM/db2forzosdeveloperextension-about] DEBUG NSP session stay on pause(F6) and does not switch to Continue(F5) (Issue #89)

You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>
CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.  Forward suspect email to ***@***.******@***.***> as an attachment (Click the More button, then forward as attachment).


Hi @R-github-sup<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FR-github-sup&data=05%7C01%7Crhakim%40oti.nyc.gov%7C2817b6de94ee4f8a9f6408da89e5e9d6%7C73d61799c28440228d4154cc4f1929ef%7C0%7C0%7C637973917825636018%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=%2B1XILAZmhA6FsSRN1jzc9Evw7ZnuJXG0FyPFrrU324w%3D&reserved=0>,

Thanks for reporting an issue. As I cannot recreate the issue, can you provide extension logs<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fibm.github.io%2Fdb2forzosdeveloperextension-about%2Fdocs%2Ftroubleshooting%2Fopening-an-issue-and-providing-logs&data=05%7C01%7Crhakim%40oti.nyc.gov%7C2817b6de94ee4f8a9f6408da89e5e9d6%7C73d61799c28440228d4154cc4f1929ef%7C0%7C0%7C637973917825636018%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=rMQsJL36Qy01QEm%2FP5QfAxaomh3IDA6%2FowUQfyebGWc%3D&reserved=0> for when you hit this issue?

-
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FIBM%2Fdb2forzosdeveloperextension-about%2Fissues%2F89%23issuecomment-1230642175&data=05%7C01%7Crhakim%40oti.nyc.gov%7C2817b6de94ee4f8a9f6408da89e5e9d6%7C73d61799c28440228d4154cc4f1929ef%7C0%7C0%7C637973917825636018%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=3t5B6jlDTnZKbu%2Brwz%2FD5kqGs31%2Fly%2B6KeWEAOw6vEw%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FA2UVZ5T3IFU4SDOJNY4ALTDV3TZCBANCNFSM57PR5M3A&data=05%7C01%7Crhakim%40oti.nyc.gov%7C2817b6de94ee4f8a9f6408da89e5e9d6%7C73d61799c28440228d4154cc4f1929ef%7C0%7C0%7C637973917825636018%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=t5qHVBKzXeNJNM0094eZx08Zi5yVHkT5txtgtrxkAMo%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

________________________________

This email, including any attachments, may contain confidential, privileged, or otherwise legally protected information intended solely for the person(s) or entity(ies) to which it is addressed. If you are not the intended recipient, you are hereby notified that any dissemination, distribution, copying, or other use of the email or its attachments is prohibited. Please immediately notify the sender of your access to the email or its attachments by replying to the message and delete all copies.

**Comment by R-github-sup** [comment]
It seems there is a firewall preventing the debug session from progressing. 
Q. the session manager port is set to default 4555. Is this the only port need to be defined in firewall and what direction inbound or outbound or both directions relative to mainframe?

**Comment by katsoohoo** [comment]
Hi @R-github-sup,

The session manager port is bidirectional with the mainframe. I also noticed in the logs that the session manager started on port `45555` rather than the default `4555`.

**Comment by MartinGuidoJensen** [comment]
I have the same problem. It seems like the wrong IP address is sent to the mainframe. My pc is connected thru cisco VPN, on the mainframe I can see that VS code has sent the wrong IP to the host, so the WLM Task is waiting for a respons that never reach the pc. How does VS determind what IP address to use?

**Comment by katsoohoo** [comment]
@MartinGuidoJensen The extension will attempt to launch debugger session manager on your local IP which will be used to communicate with the Db2 server. If this doesn't seem to be the case, can you also provide your extension's logs?

**Comment by MartinGuidoJensen** [comment]
Hi Katherine

Thanks for looking into this 😊

The only file with content I have attached.

In it I see IP: 192.168.0.164

But my “correct” IP to the Host is 10.32.211.149

I have also attached output from an ipconfig cmd

BR
Martin G. Jensen


Martin G. Jensen
Chefapplikationsarkitekt
Mainframe
Direkte: +4544555124
Mobil: +4551382992

Nykredit
Tankedraget 25
9000 Aalborg
Tlf.: 70 10 90 00

Fra: Katherine Soohoo ***@***.***>
Sendt: 24. oktober 2022 21:22
Til: IBM/db2forzosdeveloperextension-about ***@***.***>
Cc: Martin G. Jensen ***@***.***>; Mention ***@***.***>
Emne: Re: [IBM/db2forzosdeveloperextension-about] DEBUG NSP session stay on pause(F6) and does not switch to Continue(F5) (Issue #89)

Du får ikke ofte mails fra ***@***.*** Få mere at vide om, hvorfor dette er vigtigt<https://aka.ms/LearnAboutSenderIdentification>

@MartinGuidoJensen<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMartinGuidoJensen&data=05%7C01%7Cmgj%40nykredit.dk%7C73a494285585458378c908dab5f4fc55%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638022361065990369%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=kITEyDvqR%2BxJ2onfm7RNS4yR5FGe78QaYeSKrYyd65M%3D&reserved=0> The extension will attempt to launch debugger session manager on your local IP which will be used to communicate with the Db2 server. If this doesn't seem to be the case, can you also provide your extension's logs?

—
Reply to this email directly, view it on GitHub<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FIBM%2Fdb2forzosdeveloperextension-about%2Fissues%2F89%23issuecomment-1289488523&data=05%7C01%7Cmgj%40nykredit.dk%7C73a494285585458378c908dab5f4fc55%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638022361065990369%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=cl4VQxoKrdOv6pXih9lFfHfexBwE%2Fi4%2B0A2qgqNJRw4%3D&reserved=0>, or unsubscribe<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FA3XCHWWHNYEU22EPIII4XYTWE3OUNANCNFSM57PR5M3A&data=05%7C01%7Cmgj%40nykredit.dk%7C73a494285585458378c908dab5f4fc55%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638022361065990369%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Et56b6GOpHdCAhWDHseTzCXhDTd6ptfzBR%2FRSFRYtsQ%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>


Windows IP Configuration


Ethernet adapter Ethernet 2:

   Connection-specific DNS Suffix  . : nkd01100.nykreditnet.net
   Link-local IPv6 Address . . . . . : fe80::3925:63e1:9829:9b55%17
   Link-local IPv6 Address . . . . . : fe80::b479:2e68:e417:a92b%17
   IPv4 Address. . . . . . . . . . . : 10.32.211.149
   Subnet Mask . . . . . . . . . . . : 255.255.248.0
   Default Gateway . . . . . . . . . : ::
                                       10.32.208.1

Ethernet adapter Ethernet:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . : 

Ethernet adapter Ethernet 3:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . : 

Wireless LAN adapter LAN-forbindelse* 2:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . : 

Wireless LAN adapter LAN-forbindelse* 1:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . : 

Mobile Broadband adapter Mobiltelefon:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . : 

Wireless LAN adapter Wi-Fi:

   Connection-specific DNS Suffix  . : My_net
   Link-local IPv6 Address . . . . . : fe80::1153:567b:5299:db6a%38
   IPv4 Address. . . . . . . . . . . : 192.168.0.164
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : 192.168.0.1

Ethernet adapter Bluetooth-netv‘rksforbindelse:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . :

**Comment by MartinGuidoJensen** [comment]
Hi Katherine

Here it is again. I have removed the ipinfo, hope it works now.

BR
Martin G. Jensen


Martin G. Jensen
Chefapplikationsarkitekt
Mainframe
Direkte: +4544555124
Mobil: +4551382992

Nykredit
Tankedraget 25
9000 Aalborg
Tlf.: 70 10 90 00

Fra: Martin G. Jensen
Sendt: 25. oktober 2022 08:49
Til: IBM/db2forzosdeveloperextension-about ***@***.***>
Emne: SV: [IBM/db2forzosdeveloperextension-about] DEBUG NSP session stay on pause(F6) and does not switch to Continue(F5) (Issue #89)

Hi Katherine

Thanks for looking into this 😊

The only file with content I have attached.

In it I see IP: 192.168.0.164

But my “correct” IP to the Host is 10.32.211.149

I have also attached output from an ipconfig cmd

BR
Martin G. Jensen


Martin G. Jensen
Chefapplikationsarkitekt
Mainframe
Direkte: +4544555124
Mobil: +4551382992

Nykredit
Tankedraget 25
9000 Aalborg
Tlf.: 70 10 90 00

Fra: Katherine Soohoo ***@***.******@***.***>>
Sendt: 24. oktober 2022 21:22
Til: IBM/db2forzosdeveloperextension-about ***@***.******@***.***>>
Cc: Martin G. Jensen ***@***.******@***.***>>; Mention ***@***.******@***.***>>
Emne: Re: [IBM/db2forzosdeveloperextension-about] DEBUG NSP session stay on pause(F6) and does not switch to Continue(F5) (Issue #89)

Du får ikke ofte mails fra ***@***.******@***.***>. Få mere at vide om, hvorfor dette er vigtigt<https://aka.ms/LearnAboutSenderIdentification>

@MartinGuidoJensen<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMartinGuidoJensen&data=05%7C01%7Cmgj%40nykredit.dk%7C73a494285585458378c908dab5f4fc55%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638022361065990369%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=kITEyDvqR%2BxJ2onfm7RNS4yR5FGe78QaYeSKrYyd65M%3D&reserved=0> The extension will attempt to launch debugger session manager on your local IP which will be used to communicate with the Db2 server. If this doesn't seem to be the case, can you also provide your extension's logs?

**Comment by MartinGuidoJensen** [comment]
[6-Db2 Developer Extension.log](https://github.com/IBM/db2forzosdeveloperextension-about/files/9879441/6-Db2.Developer.Extension.log)

**Comment by MartinGuidoJensen** [comment]
I have dropped the file. Seems like the mail do not.
BR
Martin

**Comment by b-tsao** [comment]
Hi Martin, thank you for supplying the log file, can I get a couple more details from you?

I would like to see the `ipconfig` output with the vpn connected to confirm the IP address present on the system and in the same network session if you could run the same steps to generate the problem and supply that log file that'd be of great help, thank you!

**Comment by MartinGuidoJensen** [comment]
Hi

That's exactly what I have done. From the ipconfig output you can see that pc at that time had 2 IP addresses.

One to my local router 192.168.0.164 and one to the company network thru vpn 10.32.211.149

And in the logfile from Visual Code you can see that it is the first 192.168.0.164 that has been picked up and used.
It should have the second - Company network address that should have been used.

If it is difficult to determine this in the product I suggest to add an option where I can supply the desired ip to use.

Regards
Martin G. Jensen


Martin G. Jensen
Chefapplikationsarkitekt
Mainframe
Direkte: +4544555124
Mobil: +4551382992

Nykredit
Tankedraget 25
9000 Aalborg
Tlf.: 70 10 90 00

Fra: b-tsao ***@***.***>
Sendt: 4. november 2022 19:37
Til: IBM/db2forzosdeveloperextension-about ***@***.***>
Cc: Martin G. Jensen ***@***.***>; Mention ***@***.***>
Emne: Re: [IBM/db2forzosdeveloperextension-about] DEBUG NSP session stay on pause(F6) and does not switch to Continue(F5) (Issue #89)

Du får ikke ofte mails fra ***@***.*** Få mere at vide om, hvorfor dette er vigtigt<https://aka.ms/LearnAboutSenderIdentification>

Hi Martin, thank you for supplying the log file, can I get a couple more details from you?

I would like to see the ipconfig output with the vpn connected to confirm the IP address present on the system and in the same network session if you could run the same steps to generate the problem and supply that log file that'd be of great help, thank you!

-

**Comment by katsoohoo** [comment]
Hi @MartinGuidoJensen,

Thank you for working with us in providing all the necessary information. Unfortunately, we haven't been able to recreate the issue on our side. If possible, could we set up a video session some time next year to have you walk us through your setup and steps? You can send me an email to katherine.soohoo@ibm.com.

Happy Holidays!

**Comment by katsoohoo** [comment]
Hi @MartinGuidoJensen,

Thanks for the session today and providing us with the additional logs. Based on the new logs, we can see that Db2 is attempting to ping the "correct IP":
```
[Trace - 16.16.46] Received notification 'window/logMessage'.
Params: {
    "type": 3,
    "message": "Db2SqlServiceClient:process:stdout: 2023-01-05T16:16:46,791+0100 [ForkJoinPool.commonPool-worker-1] INFO : ClientUtility.dbg_PingManager: start calling call SYSPROC.DBG_PingSessionManager (?,?,?) for host 10.32.208.38 and port 4555\r\n"
}
...
[Trace - 16.16.47] Received notification 'client/log'.
Params: [
    3,
    "Db2SqlServiceClient:process:stdout: 2023-01-05T16:16:47,567+0100 [Client1Req - CLI_PING_MANAGER] INFO : CLI_PING_MANAGER - exit - rc: 0\r\n"
]
...
[Trace - 16.16.47] Received notification 'client/log'.
Params: [
    3,
    "Db2SqlServiceClient:process:stdout: 2023-01-05T16:16:47,693+0100 [ForkJoinPool.commonPool-worker-1] INFO : ClientUtility.dbg_PingManager: end calling call SYSPROC.DBG_PingSessionManager (?,?,?) for host 10.32.208.38 and port 4555 - rc: 0\r\n"
]
```

We also noticed that error occurs during run routine:
```
[Error - 16.18.05] Db2SqlServiceClient:process:stderr: 2023-01-05T16:18:05,342+0100 [Thread-4] ERROR: [jcc][t4][2055][14191][4.29.24] Databasesystemet kan ikke acceptere nye forespørgsler, har afbrudt alle igangværende forespørgsler
eller har afbrudt denne bestemte forespørgsel pga. en uventet fejl på målsystemet. ERRORCODE=-1224, SQLSTATE=55032
...
[Error - 16.18.05] Db2SqlServiceClient:process:stderr: 2023-01-05T16:18:05,404+0100 [Thread-4] ERROR: [jcc][t4][10335][10366][4.29.24] Ugyldig funktion: Forbindelsen er lukket. ERRORCODE=-4470, SQLSTATE=08003
```

Some additional information we would like:
* Could you also provide us with the NSP DDL and the Db2 for z/OS version you are running against?
* Are you able to perform the [run NSP action](https://ibm.github.io/db2forzosdeveloperextension-about/docs/working-with-stored-procedures/deploying-running-and-debugging-stored-procedures-basics#running-a-stored-procedure) successfully?
* Can you verify if the connection ping from Db2 was successful as indicated in the extension logs? We're wondering if the connection timeout error from the WLM logs was caused by the run routine failing.

**Comment by MartinGuidoJensen** [comment]
Hi Katherine

Shall I understand this as you don't think I am right my saying that wrong IP is sent to debug WLM?

I have tried again this morning, and on the mainframe I did a netstat and see this:
MGJ:/u/mgj: >netstat -a ! grep -i SYN
DB23WLMD 0EA4360A 10.125.100.5..4415     192.168.0.164..4555    SynSent
MGJ:/u/mgj: >

I think this proves my point that WLM thinks it is talking to 192.168.0.164 (my local IP address) It should have been the one starting with 10.32.

Here is my test NSP:
CREATE PROCEDURE "P9999MGJ" ()
VERSION V1
    ISOLATION LEVEL CS
    RESULT SETS 1
    LANGUAGE SQL
P1: BEGIN
    -- Declare cursor
    DECLARE cursor1 CURSOR WITH RETURN FOR
    -- #######################################################################
    -- # Replace the SQL statement with your statement.
    -- #  Note: Be sure to end statements with the terminator character
    -- #  (usually ';')
    -- # The example SQL statement SELECT NAME FROM SYSIBM.SYSTABLES
    -- # returns all table names from SYSIBM.SYSTABLES.
    -- ######################################################################
   SELECT NAME FROM SYSIBM.SYSTABLES;
    -- Cursor left open for client application
    OPEN cursor1;
END P1


It works fine when I do a RUN without debug.

Version(s):
*** BEGIN DISPLAY OF GROUP(DSNDB23 ) CATALOG LEVEL(V12R1M509)
                  CURRENT FUNCTION LEVEL(V12R1M508)
                  HIGHEST ACTIVATED FUNCTION LEVEL(V12R1M508)
                  HIGHEST POSSIBLE FUNCTION LEVEL(V12R1M510)
                  PROTOCOL LEVEL(2)
                  GROUP ATTACH NAME(DB23)

BR
Martin G. Jensen


Martin G. Jensen
Chefapplikationsarkitekt
Mainframe
Direkte: +4544555124
Mobil: +4551382992

Nykredit
Tankedraget 25
9000 Aalborg
Tlf.: 70 10 90 00

Fra: Katherine Soohoo ***@***.***>
Sendt: 5. januar 2023 18:10
Til: IBM/db2forzosdeveloperextension-about ***@***.***>
Cc: Martin G. Jensen ***@***.***>; Mention ***@***.***>
Emne: Re: [IBM/db2forzosdeveloperextension-about] DEBUG NSP session stay on pause(F6) and does not switch to Continue(F5) (Issue #89)

Du får ikke ofte mails fra ***@***.******@***.***>. Få mere at vide om, hvorfor dette er vigtigt<https://aka.ms/LearnAboutSenderIdentification>

Hi @MartinGuidoJensen<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMartinGuidoJensen&data=05%7C01%7Cmgj%40nykredit.dk%7C6ad23fe602c642bb6e6c08daef3fb8e9%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638085354218622524%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=IwZ1964Xx5IGTR29gVs%2FKsvmCzsj%2BVi2PazkSYeleQ8%3D&reserved=0>,

Thanks for the session today and providing us with the additional logs. Based on the new logs, we can see that Db2 is attempting to ping the "correct IP":

[Trace - 16.16.46] Received notification 'window/logMessage'.



Params: {



    "type": 3,



    "message": "Db2SqlServiceClient:process:stdout: 2023-01-05T16:16:46,791+0100 [ForkJoinPool.commonPool-worker-1] INFO : ClientUtility.dbg_PingManager: start calling call SYSPROC.DBG_PingSessionManager (?,?,?) for host 10.32.208.38 and port 4555\r\n"



}



...



[Trace - 16.16.47] Received notification 'client/log'.



Params: [



    3,



    "Db2SqlServiceClient:process:stdout: 2023-01-05T16:16:47,567+0100 [Client1Req - CLI_PING_MANAGER] INFO : CLI_PING_MANAGER - exit - rc: 0\r\n"



]



...



[Trace - 16.16.47] Received notification 'client/log'.



Params: [



    3,



    "Db2SqlServiceClient:process:stdout: 2023-01-05T16:16:47,693+0100 [ForkJoinPool.commonPool-worker-1] INFO : ClientUtility.dbg_PingManager: end calling call SYSPROC.DBG_PingSessionManager (?,?,?) for host 10.32.208.38 and port 4555 - rc: 0\r\n"



]



We also noticed that error occurs during run routine:

[Error - 16.18.05] Db2SqlServiceClient:process:stderr: 2023-01-05T16:18:05,342+0100 [Thread-4] ERROR: [jcc][t4][2055][14191][4.29.24] Databasesystemet kan ikke acceptere nye forespørgsler, har afbrudt alle igangværende forespørgsler



eller har afbrudt denne bestemte forespørgsel pga. en uventet fejl på målsystemet. ERRORCODE=-1224, SQLSTATE=55032



...



[Error - 16.18.05] Db2SqlServiceClient:process:stderr: 2023-01-05T16:18:05,404+0100 [Thread-4] ERROR: [jcc][t4][10335][10366][4.29.24] Ugyldig funktion: Forbindelsen er lukket. ERRORCODE=-4470, SQLSTATE=08003



Some additional information we would like:

  *   Could you also provide us with the NSP DDL and the Db2 for z/OS version you are running against?
  *   Are you able to perform the run NSP action<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fibm.github.io%2Fdb2forzosdeveloperextension-about%2Fdocs%2Fworking-with-stored-procedures%2Fdeploying-running-and-debugging-stored-procedures-basics%23running-a-stored-procedure&data=05%7C01%7Cmgj%40nykredit.dk%7C6ad23fe602c642bb6e6c08daef3fb8e9%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638085354218622524%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=JI5bqoZSQ0JrK49zzQacZLonPNPd%2F1lwbHYrsT8ejvs%3D&reserved=0> successfully?
  *   Can you verify if the connection ping from Db2 was successful as indicated in the extension logs? We're wondering if the connection timeout error from the WLM logs was caused by the run routine failing.

-

**Comment by katsoohoo** [comment]
Hi @MartinGuidoJensen,

Thanks for the additional logs. Based on the extension logs, we do think that the mainframe does an initial ping (using SYSPROC.DBG_PingSessionManager) correctly against your VPN IP 10.32.xx. However, as debugging continues, the mainframe incorrectly tries to communicate with your local IP 192.168.xx (as seen in WLM logs and netstat).

We have a potential fix you can download at [db2forzosdeveloperextension-1.4.0_issue89.vsix](https://github.com/IBM/db2forzosdeveloperextension-about/blob/tempfix/db2forzosdeveloperextension-1.4.0_issue89.vsix). Can you try [installing the VSIX file](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-from-a-vsix) and let us know how it goes?

Thank you!

**Comment by MartinGuidoJensen** [comment]
Hi Katherine

With this fix I am able to debug my Procedure when working from Home 😊

Thanks.

BR
Martin G. Jensen


Martin G. Jensen
Chefapplikationsarkitekt
Mainframe
Direkte: +4544555124
Mobil: +4551382992

Nykredit
Tankedraget 25
9000 Aalborg
Tlf.: 70 10 90 00

Fra: Katherine Soohoo ***@***.***>
Sendt: 10. januar 2023 01:09
Til: IBM/db2forzosdeveloperextension-about ***@***.***>
Cc: Martin G. Jensen ***@***.***>; Mention ***@***.***>
Emne: Re: [IBM/db2forzosdeveloperextension-about] DEBUG NSP session stay on pause(F6) and does not switch to Continue(F5) (Issue #89)


Hi @MartinGuidoJensen<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMartinGuidoJensen&data=05%7C01%7Cmgj%40nykredit.dk%7Ca3696164b63f4a3966f508daf29eee2d%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638089061673621891%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=AWvla99SApvF0Ba9vygVxE7E8Tnr0z22suSSFBBzggg%3D&reserved=0>,

Thanks for the additional logs. Based on the extension logs, we do think that the mainframe does an initial ping (using SYSPROC.DBG_PingSessionManager) correctly against your VPN IP 10.32.xx. However, as debugging continues, the mainframe incorrectly tries to communicate with your local IP 192.168.xx (as seen in WLM logs and netstat).

We have a potential fix you can download at db2forzosdeveloperextension-1.4.0_issue89.vsix<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FIBM%2Fdb2forzosdeveloperextension-about%2Fblob%2Ftempfix%2Fdb2forzosdeveloperextension-1.4.0_issue89.vsix&data=05%7C01%7Cmgj%40nykredit.dk%7Ca3696164b63f4a3966f508daf29eee2d%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638089061673621891%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=%2BXF6Igf4vTgrCtnapCsvRIcHmVFbh%2F4frnYPSIYnpVo%3D&reserved=0>. Can you try installing the VSIX file<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcode.visualstudio.com%2Fdocs%2Feditor%2Fextension-marketplace%23_install-from-a-vsix&data=05%7C01%7Cmgj%40nykredit.dk%7Ca3696164b63f4a3966f508daf29eee2d%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638089061673621891%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=JvLrbZMqFXct7nOFL30tObNwB0g0dlnidGlJCkRAuok%3D&reserved=0> and let us know how it goes?

Thank you!

**Comment by katsoohoo** [comment]
Thanks for verifying the fix @MartinGuidoJensen. We will include the fix in our upcoming release.

**Comment by katsoohoo** [comment]
Hi @MartinGuidoJensen,

We've implemented the fix in v2.1.0. Please give it a try when you have some time. Thanks!

**Comment by MartinGuidoJensen** [comment]
Hi Katherine

I still works in 2.1 😊 Seems like the logon to DB sometimes needs more than one 1 try to “go thru”…

Regards
Martin G. Jensen


Martin G. Jensen
Chefapplikationsarkitekt
Mainframe
Direkte: +4544555124
Mobil: +4551382992

Nykredit
Tankedraget 25
9000 Aalborg
Tlf.: 70 10 90 00

Fra: Katherine Soohoo ***@***.***>
Sendt: 24. februar 2023 19:07
Til: IBM/db2forzosdeveloperextension-about ***@***.***>
Cc: Martin G. Jensen ***@***.***>; Mention ***@***.***>
Emne: Re: [IBM/db2forzosdeveloperextension-about] DEBUG NSP session stay on pause(F6) and does not switch to Continue(F5) (Issue #89)


Hi @MartinGuidoJensen<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMartinGuidoJensen&data=05%7C01%7Cmgj%40nykredit.dk%7C7dd62eeabb2c48beae6f08db1691e3f7%7Cb05b20f7d701422a8295d38ab501738b%7C0%7C0%7C638128588078852182%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=nJomTa63WX9l8aO9Y3awkPQIQkao2zkTKYxYTqvr16g%3D&reserved=0>,

We've implemented the fix in v2.1.0. Please give it a try when you have some time. Thanks!

**Comment by katsoohoo** [comment]
Hi @MartinGuidoJensen,

Thanks for verifying the fix! For the other issue where Db2 needs to be logged in more than once, please open a new issue with more details and we can take a look at it.
